src/app/login/oauth2/code/google/page.tsx

```diff
@@ -1,28 +1,41 @@
 'use client';
-import React, { useEffect, useState } from 'react';
-import { useSearchParams } from 'next/navigation';
+import React, { useEffect } from 'react';
+import { useRouter, useSearchParams } from 'next/navigation';
 import type { AxiosResponse } from 'axios';
+import { useAtom } from 'jotai';
+import Cookies from 'js-cookie';
 
 import { api } from '@/apis';
+import { memberAtom } from '@/features/member/atoms';
 
 const GoogleLogin = () => {
-  const [email, setEmail] = useState<string>('');
   const searchParams = useSearchParams();
+  const router = useRouter();
   const token = searchParams.get('token');
-  // TODO: query string으로 JWT Token을 받아서 쿠키에 저장하는 로직이 필요합니다.
+  const [_, setMember] = useAtom(memberAtom);
 
   useEffect(() => {
-    try {
-      api.get('/my').then((response: AxiosResponse) => {
-        setEmail(response.data.email);
-        console.log(response.data.email);
-      });
-    } catch (error) {
-      console.error(error);
+    console.log('token', token);
+    if (token) {
+      Cookies.set('accessToken', token, { expires: 7 });
+
+      try {
+        api.get('/my').then((response: AxiosResponse) => {
+          if (!response.data.username) {
+            setMember(response.data);
+            router.push('/member/new/nickname');
+          } else {
+            setMember(response.data);
+            router.push('/home');
+          }
```

Comment:
`setMemeber`가 공통되니 밖으로 빼서
```suggestion
setMember(response.data);
router.push(response.data.username ? '/home' : '/member/new/nickname');
```
요런 형태의 코드는 어떨까요??

Reply:
반영 완료~~!! 준영 최고